Orange/widgets/visualize/owruleviewer.py

```diff
@@ -0,0 +1,251 @@
+import numpy as np
+
+from Orange.data import Table
+from Orange.widgets import widget, gui, settings
+from Orange.widgets.utils.itemmodels import PyTableModel
+
+from PyQt4.QtCore import Qt, QSize, pyqtSignal, QRectF, QLineF, QLocale
+from PyQt4.QtGui import (QApplication, QLabel, QTableView, QStandardItem,
+                         QStandardItemModel, QSortFilterProxyModel,
+                         QMainWindow, QMouseEvent, QGraphicsView,
+                         QPen, QBrush, QColor, QAbstractItemView,
+                         QItemDelegate, QStyledItemDelegate, QPainter,
+                         QFontMetrics, QStyle)
+
+
+class Rule:  # temp - DELETE
+    curr_class_dist = [0, 28, 0]
+
+    def __str__(self):
```

Comment:
You can put arbitrary objects into PyTableModel. Yet, if you want their `__str__()` values to be used for Qt.DisplayRole, you need to extend the model like so:
```py
class MyTableModel(PyTableModel):
    def data(index, role=Qt.DisplayRole):
        value = super().data(index, role)
        return str(value) if role == Qt.DisplayRole else value
```
Otherwise, the view will just show blanks as it doesn't know how to convert arbitrary objects into strings.